### PR TITLE
[joystick] Remove Bluetooth MAC address from device name

### PR DESCRIPTION
**Pull request by fetzerch**
This is the alternative version of https://github.com/kodi-game/peripheral.joystick/pull/17 using PCRE instead of C++11 regex. Only tested on Linux currently.
I guess it needs to manage the PCRE dependency somehow on other platforms but I haven't yet figured out how that works.

### CODE REVIEW

**Comment by Montellese** [comment]
Dependencies need to be defined in the `depends` directory, see e.g. https://github.com/kodi-pvr/pvr.argustv/tree/master/depends/common/jsoncpp for how `jsoncpp` is defined as a dependency.

**Comment by fetzerch** [comment]
Ok, @garbear: I started already in a separate branch https://github.com/fetzerch/peripheral.joystick/commit/9710f6d2d614b8fba59b24533a621bd5ca70d4fa. PCRE ships it's own CMakeLists.txt which compiles flawlessly in Linux. Wanted to try it also under Windows, but my old setup is still VS2010 based which doesn't work with binary addons anymore. So I'll have to upgrade that first tomorrow. Until then: is it ok to use the upstream CMakeLists.txt (which builds also tests and utils), or is it preferred to ship a custom made CMakeLists.txt that only compiles a static lib.

**Comment by garbear** [comment]
If upstream CMakeLists.txt is provided, let's use that

**Comment by fetzerch** [comment]
Allright. One more comment on the change itself (after testing with OpenELEC). This change is only needed when connecting the controller through QtSixA (sixpair, sixad). This is typically the case on Ubuntu (which doesn't have Bluez 5 or has the sixaxis plugin disabled). When using Bluez 5 (default on OpenELEC) the controller doesn't have the BT MAC address in the name. (will push another keymap in a sec for that). I'm wondering now if we should make that change Linux only. Or if we should keep it generic as it won't hurt and it could also be that there are some controllers or drivers with that behaviour on Windows or Mac as well.

**Comment by garbear** [comment]
Generic please

File: src/utils/StringUtils.cpp [hunk 1]
```diff
@@ -62,6 +63,12 @@ std::string StringUtils::MakeSafeUrl(const std::string& str)
   return safeUrl;
 }
 
+std::string& StringUtils::RemoveMACAddress(std::string& str)
+{
+  pcrecpp::RE re("[\\(\\[]?([0-9A-Fa-f]{2}[:-]){5}([0-9A-Fa-f]{2})[\\)\\]]?");
+  re.GlobalReplace("", &str);
+}
```
Comment (garbear) [inline src/utils/StringUtils.cpp]: return str?

**Comment by Montellese** [comment]
AFAIK the static lib used on win32 with Kodi has been built with the cmake buildsystem provided by PCRE. So I don't think you need to provide your own.

File: src/utils/StringUtils.cpp (hunk 1 above)
Comment (fetzerch) [inline src/utils/StringUtils.cpp]: uups. thx, fixed

**Comment by fetzerch** [comment]
@garbear, @Montellese: Thanks for your help. The change now successfully compiles in linux and windows (vs12). Can any of you try osx?

Windows was a bit tricky, as on windows it's required to link against pcre.lib specifically and define PCRE_STATIC. In Kodi this is all hidden in preprocessor magic in the regex abstraction (see https://github.com/xbmc/xbmc/blob/f42c66f130805765b0c30d947980d5f2af847cde/xbmc/utils/RegExp.h#L32). With CMake we can do all that in the FindPCRE.cmake :).

File: cmake/FindPCRE.cmake [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  find_library(PCRECPP_LIBRARY NAMES pcrecpp pcrecppd)
+  find_library(PCRE_LIBRARY NAMES pcre pcred)
+endif()
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(PCRE DEFAULT_MSG PCRE_INCLUDE_DIRS PCRECPP_LIBRARY PCRE_LIBRARY)
+mark_as_advanced(PCRE_INCLUDE_DIRS PCRECPP_LIBRARY PCRE_LIBRARY)
+if(PCRE_FOUND)
+  set(PCRE_LIBRARIES ${PCRECPP_LIBRARY} ${PCRE_LIBRARY})
+  set(PCRE_DEFINITIONS "-DPCRE_STATIC=1")
```
Comment (Montellese) [inline cmake/FindPCRE.cmake]: Should this be made win32 specific? Or is it required for Linux as well?

Comment (fetzerch) [inline cmake/FindPCRE.cmake]: done (no required for linux)

**Comment by garbear** [comment]
This compiles successfully in OSX. Ready to go?

**Comment by fetzerch** [comment]
Not sure if that's a question to me. From my side it's ready unless you want me to change anything. Works well on Linux, and doesn't cause any regressions on Windows.

**Pull request merged**